src/services/companyService.ts

```diff
@@ -10,6 +12,13 @@ export const getCompanies = (
   containerName: NonEmptyString,
   blobName: NonEmptyString
 ) =>
-  getBlobData(blobServiceClient, containerName, blobName, UsersCompanies)
-    .map(_ => _.find(elem => elem.fiscalCode === fiscalCode))
-    .map(result => fromNullable(result).map(_ => _.companies));
+  pipe(
+    getBlobData(blobServiceClient, containerName, blobName, UsersCompanies),
+    TE.map(_ => _.find(elem => elem.fiscalCode === fiscalCode)),
+    TE.map(
+      flow(
+        O.fromNullable,
+        O.map(_ => _.companies)
```

Comment:
I guess you can use:

```suggestion
    TE.map(A.findFirstMap(elem => elem.fiscalCode === fiscalCode ? 
      O.some(elem.companies) : O.none),
```